### PR TITLE
Color blending fixes

### PR DESCRIPTION
**Pull request by jac8888**
The gradient code had the same rounding error in the color lerp that the border radius code did. I also fixed an error in the way the blending is done in the animation code. (More details [here](https://github.com/mikke89/RmlUi/discussions/207).)

### CODE REVIEW

File: Source/Core/DecoratorGradient.cpp [hunk 1]
```diff
@@ -95,16 +95,39 @@ DecoratorDataHandle DecoratorGradient::GenerateElementData(Element* element) con
...
+			};
+			vertices[i].colour = color;
+
 		}
 	}
 	else if (dir == Direction::Vertical)
 	{
 		for (int i = 0; i < (int)vertices.size(); i++)
 		{
 			const float t = (vertices[i].position.y - padding_offset.y) / padding_size.y;
```
Comment (mikke89) [inline Source/Core/DecoratorGradient.cpp:118]: Should we clamp this as well perhaps?

**Review by mikke89 — COMMENTED** [review]
Thanks a lot! These are some great changes.

Could you perhaps move all the color interpolations into a separate function? They are quite verbose, and especially now that they are used several places.

**Comment by jac8888** [comment]
Sure thing; would you say that the lerp should be overloaded for Colourb to be like this?
Will add the clamp as well.

**Comment by mikke89** [comment]
Great. I think it's better to make a separate function for this one.

**Comment by jac8888** [comment]
I do think it makes sense to overload the lerp since it doesn't behave in the correct way for Colourbs (truncating rather than rounding the result leading to errors), and someone is likely to use the wrong function unknowingly if there is another one. I'm not really sure where else it would make sense to go (locally, another function in Math)?

**Comment by mikke89** [comment]
I figured it would be easier to separate them in case we later provide a way to select different color interpolation strategies. We don't really have a better place right now than math, that should be fine.

We could do a static assert to make sure people don't use lerp with Colourb.

**Comment by jac8888** [comment]
All right, I've made those changes. :)

**Comment by mikke89** [comment]
Awesome work! Thank you! :)

**Pull request merged**